src/wasm-lib/kcl/src/std/extrude.rs

```diff
@@ -46,16 +46,92 @@ async fn inner_extrude(length: f64, sketch_group: Box<SketchGroup>, args: Args)
     )
     .await?;
 
-    Ok(Box::new(ExtrudeGroup {
+    // TODO maybe this should loop through until it gets a edge_id
+    let first_segment = &sketch_group.value[0];
+    let edge_id = match first_segment {
+        Path::ToPoint { base } => base.geo_meta.id,
+        // Handle other variants if necessary
+        _ => {
+            return Err(KclError::Type(KclErrorDetails {
+                message: "Expected a Path::ToPoint variant".to_string(),
+                source_ranges: vec![args.source_range],
+            }))
+        }
+    };
+
+    let solid3d_info = args
+        .send_modeling_cmd(
+            id,
+            kittycad::types::ModelingCmd::Solid3DGetExtrusionFaceInfo {
+                edge_id,
+                object_id: sketch_group.id,
+            },
+        )
+        .await?;
+
+    let face_infos = if let kittycad::types::OkWebSocketResponseData::Modeling {
+        modeling_response: kittycad::types::OkModelingCmdResponse::Solid3DGetExtrusionFaceInfo { data },
+    } = solid3d_info
+    {
+        data.faces
+    } else {
+        return Err(KclError::Type(KclErrorDetails {
+            message: "Invalid solid3dInfo format".to_string(),
+            source_ranges: vec![args.source_range],
+        }));
+    };
+
+    // Create a hashmap for quick id lookup
+    let mut face_id_map = std::collections::HashMap::new();
+    let mut start_cap_id = None;
+    let mut end_cap_id = None;
+
+    for face_info in face_infos {
+        match face_info.cap {
+            kittycad::types::ExtrusionFaceCapType::Bottom => start_cap_id = face_info.face_id,
+            kittycad::types::ExtrusionFaceCapType::Top => end_cap_id = face_info.face_id,
+            _ => {
+                if let Some(curve_id) = face_info.curve_id {
+                    face_id_map.insert(curve_id, face_info.face_id);
+                }
+            }
+        }
+    }
+
+    // Iterate over the sketch_group.value array and add face_id to GeoMeta
+    let mut new_value: Vec<ExtrudeSurface> = Vec::new();
+    for path in sketch_group.value.iter() {
+        if let Some(Some(actual_face_id)) = face_id_map.get(&path.get_base().geo_meta.id) {
+            let extrude_surface = ExtrudeSurface::ExtrudePlane {
+                position: sketch_group.position, // TODO should be for the extrude surface
+                rotation: sketch_group.rotation, // TODO should be for the extrude surface
+                face_id: *actual_face_id,
```

Comment:
@Irev-Dev I am confused if we have this here why we also need it in geo_meta? mostly trying to avoid geo_meta having specific data versus generic data esp if its elsewhere in the memory item anyways